src/context/AppContextProvider.tsx

```diff
@@ -46,6 +46,7 @@ export const AppContextProvider = ({
   progressStore: TextStreamStore;
   messageHistoryStore: MessageHistoryStore;
 }) => {
+  const [isOperationValidated, setIsOperationValidated] = useState(false);
```

Comment:
I played around with a more OOP-approach of handling whether to render the `inProgressComponent` within the class itself, but it involved much more code than simple react state.